src/uwtools/config/formats/ini.py

```diff
@@ -81,13 +81,9 @@ def dump_dict(path: OptionalPath, cfg: dict, opts: Optional[ns] = None) -> None:
         parser = configparser.ConfigParser()
         s = StringIO()
         cfgdepth = depth(cfg)
-        assert cfgdepth in (1, 2)  # 1 => .sh, 2 => .ini
-        if cfgdepth == 2:
-            parser.read_dict(cfg)
-            parser.write(s, space_around_delimiters=opts.space if opts else True)
-        else:
-            for key, value in cfg.items():
-                print(f"{key}={value}", file=s)
+        assert cfgdepth in (1, 2)  # 2 => .ini
```

Comment:
So I had done that, but switched back since INI can technically take 1 or 2 levels. However, given that I'm going to have to make the output depth controlled, I definitely like the idea of overriding to let it define a `MAXDEPTH` I can interrogate later!

Reply:
I wasn't aware that INI can take 1 level. Can you show an example?

Reply:
Strictly, afaik, the INI format doesn't require a header per se, although `ConfigParser` does. I think you're right to say that there's no point in supporting a special case though, or more in enforcing what we want to see. I'll clean that one up

Reply:
Going through a few times, we do kind of bluntly check depth compatibility, but a more rigorous treatment would be good for UW-420. I'm also not sure how I feel about currently having namelists fully incompatible - I would like to be able to convert them personally, but I don't see an actual need for it, so its probably ok to keep here for now.